src/browser/ui/dom/components/ReactDOMOption.js

```diff
@@ -86,7 +87,22 @@ var ReactDOMOption = ReactClass.createClass({
       props = assign({}, props, {selected: this.state.selected});
     }
 
-    return option(props, this.props.children);
+    var content = '';
+
+    // Flatten children and warn if they aren't strings or numbers;
+    // invalid types are ignored.
+    ReactChildren.forEach(this.props.children, function(child) {
+      if (typeof child === 'string' || typeof child === 'number') {
```

Comment:
Can you make sure this skips over null/undefined children without a warning and add a test for that?